core/src/main/java/org/elasticsearch/cluster/action/shard/ShardStateAction.java

```diff
@@ -234,51 +232,50 @@ public void clusterStateProcessed(String source, ClusterState oldState, ClusterS
         }
 
         @Override
-        public String describeTasks(List<ShardRoutingEntry> tasks) {
-            return tasks.stream().map(entry -> entry.getShardRouting().toString()).reduce((s1, s2) -> s1 + ", " + s2).orElse("");
-        }
-
-        @Override
         public BatchResult<ShardRoutingEntry> execute(ClusterState currentState, List<ShardRoutingEntry> tasks) throws Exception {
             BatchResult.Builder<ShardRoutingEntry> batchResultBuilder = BatchResult.builder();
+            List<ShardRoutingEntry> validTasks = new ArrayList<>();
+            List<FailedRerouteAllocation.FailedShard> validShardRoutings = new ArrayList<>();
 
-            // partition tasks into those that correspond to shards
-            // that exist versus do not exist
-            Map<ValidationResult, List<ShardRoutingEntry>> partition =
-                tasks.stream().collect(Collectors.groupingBy(task -> validateTask(currentState, task)));
+            for (ShardRoutingEntry task : tasks) {
+                IndexMetaData indexMetaData = currentState.metaData().index(task.shardId.getIndex());
+                if (indexMetaData == null) {
+                    // tasks that correspond to non-existent shards are marked as successful
+                    batchResultBuilder.success(task);
```

Comment:
can we add some trace logging here? I can imagine it will save some WTF at some point.